pkg/loki/loki.go

```diff
@@ -145,21 +146,20 @@ func (t *Loki) Stop() error {
 }
 
 func (t *Loki) stop(m moduleName) {
-	if _, ok := t.inited[m]; !ok {
-		return
-	}
-	delete(t.inited, m)
-
-	for _, dep := range modules[m].deps {
-		t.stop(dep)
-	}
-
-	if modules[m].stop == nil {
-		return
+	t.stopModule(m)
+	deps := orderedDeps(m)
+	// iterate over our deps in reverse order and call stopModule
+	for i := len(deps) - 1; i >= 0; i-- {
+		t.stopModule(deps[i])
 	}
+}
 
+func (t *Loki) stopModule(m moduleName) {
 	level.Info(util.Logger).Log("msg", "stopping", "module", m)
-	if err := modules[m].stop(t); err != nil {
-		level.Error(util.Logger).Log("msg", "error stopping", "module", m, "err", err)
+	if modules[m].stop != nil {
+		if err := modules[m].stop(t); err != nil {
+			level.Error(util.Logger).Log("msg", "error stopping", "module", m, "err", err)
+		}
 	}
+	delete(t.inited, m)
```

Comment:
Is this needed still?

Reply:
I guess the `inited` field can be removed completely from the Loki struct